adapter/internal/oasparser/envoyconf/envoyconf_internal_test.go

```diff
@@ -396,6 +396,30 @@ func TestGenerateRegex(t *testing.T) {
 			message:       "when the resource ends with *, trailing characters substitution passes",
 			isMatched:     false,
 		},
+		{
+			inputpath:     "/v2/pet/{petId}/images/*",
+			userInputPath: "/v2/pet/123/images/123-foo.png",
+			message:       "when the resource ends with * and path params in the middle passes",
+			isMatched:     true,
+		},
+		{
+			inputpath:     "/v2/pet/{petId}/images/*",
+			userInputPath: "/v2/pet123/images/123-foo.png",
+			message:       "when the resource ends with * and path params in the middle fails",
+			isMatched:     false,
+		},
+		{
+			inputpath:     "/v2/pet/{petId}/images)/*",
+			userInputPath: "/v2/pet/123/images)/123-foo.png",
+			message:       "when the resource ends with *, special char in the path and path params in the middle passes",
```

Comment:
Only difference in the line #420 is fails. Is that intended ?
`when the resource ends with *, special char in the path and path params in the middle passes`
`when the resource ends with *, special char in the path and path params in the middle **fails**`

same applies to the line 402 and 408 as well.

Reply:
Nope `userInputPath` is different in these two test cases. There is `)` in this path and not in the next one.